--- src/cheese-window.c.orig	2008-07-15 14:32:03.346673101 +0200
+++ src/cheese-window.c	2008-07-15 14:44:27.966701974 +0200
@@ -34,6 +34,7 @@
 #include <gst/interfaces/xoverlay.h>
 #include <gtk/gtk.h>
 #include <libebook/e-book.h>
+#include <libgnome/gnome-url.h>
 
 #include "cheese-countdown.h"
 #include "cheese-effect-chooser.h"
@@ -124,7 +125,7 @@
   GtkWidget *error_dialog;
   gboolean ret;
 
-  ret = g_app_info_launch_default_for_uri (url, NULL, &error);
+  ret = gnome_url_show (url, &error);
   if (ret == FALSE)
   {
     error_dialog = gtk_message_dialog_new (GTK_WINDOW (dialog), 
@@ -148,7 +149,7 @@
 
   uri = g_strconcat ("mailto:", email, NULL);
 
-  ret = g_app_info_launch_default_for_uri (uri, NULL, &error);
+  ret = gnome_url_show (uri, &error);
   if (ret == FALSE)
   {
     error_dialog = gtk_message_dialog_new (GTK_WINDOW (dialog), 
@@ -240,7 +241,7 @@
   uri = g_filename_to_uri (filename, NULL, NULL);
   g_free (filename);
  
-  ret = g_app_info_launch_default_for_uri (uri, NULL, &error);
+  ret = gnome_url_show (uri, &error);
   if (ret == FALSE)
   {
     dialog = gtk_message_dialog_new (GTK_WINDOW (cheese_window->window), 
@@ -532,7 +533,7 @@
   GError *error = NULL;
   gboolean ret;
 
-  ret = g_app_info_launch_default_for_uri ("ghelp:cheese", NULL, &error);
+  ret = gnome_url_show ("ghelp:cheese", &error);
 
   if (ret == FALSE) 
   {
