From: Andreas Henriksson <andreas@fatal.se>
Subject: Use LT_LIB_M to find the math library

libcheese-gtk (sometimes?) needs to link with -lm
to find the fmod function.

--- a/Makefile.am
+++ b/Makefile.am
@@ -87,6 +87,7 @@
 
 libcheese_gtk_la_LIBADD = \
 	libcheese.la \
+	$(LIBM) \
 	$(CHEESE_LIBS) \
 	$(CHEESE_GTK_LIBS)
 
--- a/configure.ac
+++ b/configure.ac
@@ -24,6 +24,9 @@
 LT_PREREQ([2.2])
 LT_INIT([disable-static])
 
+# Find an appropriate libm, for fmod()
+LT_LIB_M
+
 PKG_PROG_PKG_CONFIG([0.24])
 
 AC_SUBST([CHEESE_WFLAGS], ["-Wall -DGSEAL_ENABLE"])
